wasm/jit/jit_amd64.go

```diff
@@ -297,16 +57,26 @@ type amd64Builder struct {
 	requireFunctionCallReturnAddressOffsetResolution []*obj.Prog
 }
 
-func (b *amd64Builder) compile() ([]byte, error) {
-	code, err := mmapCodeSegment(b.builder.Assemble())
+// Implements compiler.emitPreamble for amd64Compiler.
+func (c *amd64Compiler) emitPreamble() {
+	// We assume all function parameters are already pushed onto the stack by
+	// the caller.
+	c.pushFunctionParams()
+	// Initialize the reserved registers first of all.
+	c.initializeReservedRegisters()
+}
+
+// Implements compiler.compile for amd64Compiler.
+func (c *amd64Compiler) compile() ([]byte, uint64, error) {
```

Comment:
I think godoc like this don't add value. Also IDEs warn when they don't start with the function name